src/odb/include/odb/db.h

```diff
@@ -967,6 +967,8 @@ class dbBlock : public dbObject
   ///
   dbSet<dbITerm> getITerms();
 
+  dbITerm* getITerm(int);
+
```

Comment:
rm - duplicates dbITerm::getITerm

Reply:
Turns out that the getITerm is a method in BTerm. We also need a generic method for getting the ITerm by id from the block (in absence of bterm -- for example when fishing out wiring of iterms from modnets).

Reply:
The method in dbBTerm is unrelated and this is still duplicate.

Reply:
Sorry, I don't see the duplicate. Please can you be more specific what it is duplicating ?

This call is in the class dbBlock. It is called at a point at which there is no dbITerm. Perhaps you are alluding to somehow not calling this and using some other api. There are at least 4 other calls to get the dbITerm: eg   static dbITerm* getITerm(dbBlock* block, uint oid); (in class dbInst) or getITerm in dbCapNode. etc etc. Please can you be more specific ?

Reply:
I'm not sure what the "dbNet stuff" refers to.

Reply:
dbNet.cpp

For example see: void dbNet::setDrivingITerm(int id)
{
  _dbNet* net = (_dbNet*) this;
  net->_drivingIterm = id;
}
int dbNet::getDrivingITerm()
{
  _dbNet* net = (_dbNet*) this;
  return net->_drivingIterm;
}
It is as if the authors of dbNet.cpp were using ints to represent ids and somehow using -1 as the empty case for drivingITerm -- see for example _dbNet::_dbNet.

Looks strange to me, but then am no expert. Mixing unsigneds/ints in this way seems fishy.

Reply:
Almost all uses of unit in odb are a legacy and a bad idea.  I have no idea why anyone would care about -1 vs 0 in
```
  ///
  /// Returns driving term id assigned of this net. -1 if not set, 0 if non
  /// existent
  ///
  int getDrivingITerm();
```